anvio/homogeneityindex.py

```diff
@@ -46,7 +46,7 @@ def compute_functional_index(self, gene_cluster_sequences):
         max_score = 0
         for residue_number in range(0, num_residues):
             residues = []
-            for gene_sequence in range(0, num_sequences):
+            for gene_sequence in range(0, (num_sequences)-1):
```

Comment:
I'm not sure why this was necessary. We shouldn't have to make this change since Python's way of counting items in arrays could not have changed between 3.7 to 3.10. If this part of the code is exploding after the update to 3.10, there must have been something else gone wrong I thin.

Reply:
> I just tested the pangenomics workflow after removing this change, and it seems to be working fine, @metehaansever. Do you remember why you've made this change?

I made a change here because I got a list index out-of-range error.

Reply:
Can you try again running

```
anvi-self-test --suite pangenomics -o PAN-TEST
```

after reverting this change and see if you are getting that error again? There must be something else going wrong before this.

Reply:
I receive another error right now, 

```
Config Error:: An anvi'o function needs 'average_nucleotide_identity.py' to be installed on   
              your system, but it doesn't seem to appear in your path :/ If you are certain  
              you have it on your system (for instance you can run it by typing              
              'average_nucleotide_identity.py' in your terminal window), you may want to send
              a detailed bug report. Sorry! 
```
I'll test and fix this change as soon as I fix this problem.